packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -64,6 +64,23 @@ function DocSidebarItemCategory({
     return isActive ? false : item.collapsed;
   });
 
+  const menuListRef = useRef<HTMLUListElement>(null);
+  const [menuListHeight, setMenuListHeight] = useState<string | undefined>(
```

Comment:
Of course, we can eliminate to create a new state for storing the height and set it via `menuListRef.current.style.height`, but it seems to me better to do this using React.